internal/storage/dbstorage/dbstorage.go

```diff
@@ -81,6 +101,26 @@ func (pg *PostgresDB) SetURL(shortURL string, originalURL string) (string, error
 	}
 }
 
+func (pg *PostgresDB) SetUserURL(shortURL string, originalURL string, userID int) (string, error) {
+	tx, err := pg.db.Begin()
+	if err != nil {
+		tx.Rollback()
+		return "", fmt.Errorf("failed to begin transaction: %w", err)
+	}
```

Comment:
```suggestion
	if err != nil {
		return "", fmt.Errorf("failed to begin transaction: %w", err)
	}
	defer func() {
		if err != nil {
			tx.Rollback()
		}
	}()
```
Если при `pg.db.Begin()` возникнет ошибка, то tx будет равен `nil`. В этом случае попытка вызвать от него `Rollback()` приведет к панике. А вот если в процессе обслуживания какого-либо из запросов возникнет ошибка, то нам необходимо будет сделать откат. Формально, можно дергать даже 
```
defer tx.Rollback()
```
так как, если перед ним будет выполнен Commit, то фугкция просто сработает вхолостую.